src/common/helpers/design-tokens.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import designTokensColors from '~generated/design-tokens-colors.json';
+
+export type DesignTokens = {
+  [K in keyof typeof designTokensColors]: `--${K}`;
+}[keyof typeof designTokensColors];
+
+export type DesignTokensColors = {
+  [K in DesignTokens]: K extends `--echoes-color-${string}` ? K : never;
+}[DesignTokens];
```

Comment:
For readability, I'd prefer not to have the `--` prefix in the values.
We could instead prefix it in `IconWrapper` :
```
${({ color }) => isDefined(color) && `color: var(--${color});`}
```

WDYT?

Reply:
We can easily remove it from the value, I was thinking we want all usage of the design token to look the same.
But maybe it's already not the case even with that since in CSS there is the `var()` also...
Alright I'll remove the `--` from he type.